build/generate-struct-arrays.js

```diff
@@ -175,6 +176,18 @@ createStructArrayType('feature_index', createLayout([
     { type: 'Uint16', name: 'bucketIndex' }
 ]), true);
 
+createStructArrayType('paint_buffer_offset', createLayout([
+    // the index of the feature in the original vectortile
+    { type: 'Float64', name: 'featureId' },
```

Comment:
Corresponds to `uint64_t` in native

Reply:
I still confused by this. What's the underlying type defined in the vector tile spec? A Float64 can't represent every uint64_t.

Reply:
The [vector tile spec](https://github.com/mapbox/vector-tile-spec/tree/master/2.1#42-features) doesn't define a type AFAICT. Native vt libraries use a `unint64_t` to represent ids. The vectortile.js library uses a `number` type to represent the id, which I believe has the same precision issue as `Float64` does here.

Reply:
The feature IDs in the `.proto` file of the current vector tile spec are [64-bit unsigned integers](https://github.com/mapbox/vector-tile-spec/blob/master/2.1/vector_tile.proto#L32), although this should have been mentioned in the text too. GeoJSON prefers string IDs, so for VT3 we are discussing [generalizing IDs to any representable type](https://github.com/mapbox/vector-tile-spec/pull/120) and would appreciate input on what is feasible in client code.

Reply:
Huh yeah JS's `Number.MAX_SAFE_INTEGER == 2^53 - 1` so technically that's a bug in our current implementation... although apparently we've gotten this far without hearing anyone complain about the feature ID 2^54 not working.

Reply:
@ChrisLoer that limit is for Integers. But `Number` is always a double, with `Number.MAX_VALUE == 1.79E+308` . In order to use a large Number in a typed array, as is the case here, the array type should be large enough to accommodate `uint64_t`. `Float64` is the only type that can do that here.

Reply:
Created #7020 to track the feature ID bit-size discrepancy.